src/Surfnet/ServiceProviderDashboard/Application/Assembler/ServiceStatusAssembler.php

```diff
@@ -0,0 +1,195 @@
+
+        $states = $this->getStates($service, $serviceStatusService);
+        $mappedStates = $this->mapStates($states);
+
+        $this->serviceStatusDto = new ServiceStatusDto(
+            $service->getName(),
+            $serviceLink,
+            $this->getEntities($service, $router),
+            $mappedStates
+        );
+
+    }
+
+    /**
+     * @return ServiceStatusDto
+     */
+    public function getDto()
+    {
+        return $this->serviceStatusDto;
+    }
+
+    /**
+     * @param Service $service
+     * @param ServiceStatusService $serviceStatusService
+     * @return array
+     */
+    private function getStates(Service $service, ServiceStatusService $serviceStatusService)
+    {
+        $states = [];
+        $type = $service->getServiceType();
+
+        $intakeStatus = $service->getIntakeStatus();
+        if ($intakeStatus != Service::INTAKE_STATUS_NOT_APPLICABLE) {
+            $states[self::SERVICE_STATE_INTAKE_CONDUCTED] = $intakeStatus;
+        }
+
+        $states[self::SERVICE_STATE_ENTITY_ON_TEST] = $serviceStatusService->getEntityStatus($service);
+
+        if ($type == Service::SERVICE_TYPE_NON_INSTITUTE) {
+            $states[self::SERVICE_STATE_CONTRACT_SIGNED] = $service->getContractSigned();
+        }
+
+        if ($type == Service::SERVICE_TYPE_INSTITUTE) {
+            $states[self::SERVICE_STATE_REPRESENTATIVE_APPROVED] = $service->getSurfconextRepresentativeApproved();
+        }
+
+        if ($service->isPrivacyQuestionsEnabled()) {
+            $states[self::SERVICE_STATE_PRIVACY_QUESTIONS] = $serviceStatusService->hasPrivacyQuestions($service);
+        }
+
+        $states[self::SERVICE_STATE_PRODUCTION_CONNECTION] = $service->getConnectionStatus();
+
+        return $states;
+    }
+
+    /**
+     * @param array $states
+     * @return array
+     */
+    private function mapStates($states)
+    {
+        $result = [];
+        foreach ($states as $name => $value) {
+            $result[$name] = $this->mapState($name, $value);
+        }
+        return $result;
+    }
+
+    /**
+     * @param Service $service
+     * @param RouterInterface $router
+     * @return array
+     */
+    private function getEntities(Service $service, RouterInterface $router)
+    {
+        $entityLinks = [];
+        foreach ($service->getEntities() as $entity) {
+            $url = $router->generate('entity_edit', ['id' => $entity->getId()]);
+            $entityLinks[$url] = $entity->getName();
+        }
+
+        return $entityLinks;
+    }
+
+    /**
+     * @param string $name
+     * @param mixed $value
+     * @return string
+     */
+    private function mapState($name, $value)
+    {
+        if (isset($this->stateMapping[$name])) {
+            if (isset($this->stateMapping[$name][$value])) {
+                return $this->stateMapping[$name][$value];
+            }
+        }
+
+        throw new \InvalidArgumentException(sprintf('invalid status requested: %s %s', $name, $value));
```

Comment:
Please use or create an exception from the application namespace. And secondly in this project we alway import all classnames. I've seen other instance in this PR where you use them without importing them first. Please fix this.